presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -291,7 +292,7 @@ public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, N
         public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
         {
             // A grouping() is preceded by a GroupId node when the source node is projecting a groupid symbol
-            if (!source.getOutputSymbols().contains(new Symbol("groupid")) && node.getName().toString().equals(GROUPING)) {
+            if (!source.getOutputSymbols().stream().anyMatch(symbol -> symbol.getName().contains(GROUPID_SYMBOL_HINT)) && node.getName().toString().equals(GROUPING)) {
```

Comment:
In order to search for appropriate `GroupIdNode` you could have just use `PlanNodeSearcher` class and its `findFirst` method. Sorry I didn't mentioned it earlier